source/agora/node/Registry.d

```diff
@@ -407,6 +407,33 @@ public class NameRegistry: NameRegistryAPI
         sender(reply);
     }
 
+    /***************************************************************************
+        
+        Find zone in registry
+
+        Params:
+            name = Domain name of the zone to be found
+
+        Returns:
+            Function pointer of answer for the zone, `null` is returned when no
+            zone can be found
+
+    ***************************************************************************/
+    
+    auto findZone (Domain name) @safe
+    {
+        if (this.validators.owns(name))
+            return &this.validators.answer_owns;
+        else if (this.validators.matches(name))
+            return &this.validators.answer_matches;
+        else if (this.flash.owns(name))
+            return &this.flash.answer_owns;
+        else if (this.flash.matches(name))
+            return &this.flash.answer_matches;
+        else
+            return null;
+    }
```

Comment:
This will do 4 comparisons for every non-matching zones. It feels a bit odd to compare twice.
I really think the better fix would be to have a way to bridge `owns` and `matches` since they are obvious duplication.

Reply:
New method of the zone finding will resolves this problem now and also opens a way to implement even more sub-zones in future; e.g. `verytrusted.validators.localhost` next to `validators.localhost`